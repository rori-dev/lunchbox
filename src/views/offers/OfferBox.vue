<template>
  <div
    class="flex flex-col items-center
           px-4 py-2 sm:pt-4 sm:pb-6
           shadow-xl bg-white rounded-lg
           border-l-8 sm:border-t-8 sm:border-l-0 border-primary-400"
  >
    <span class="text-3xl sm:text-4xl font-serif text-neutral-800">
      {{ provider.name }}
    </span>
    <Offer v-for="offer in offers" :key="offer.id" :offer="offer" />
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import Offer from '@/views/offers/Offer.vue'
import { LunchProvider } from '@/model/LunchProvider'
import { LunchOffer } from '@/model/LunchOffer'

@Component({
  components: {
    Offer,
  },
})
export default class OfferBox extends Vue {
  @Prop() provider!: LunchProvider

  @Prop() offers!: LunchOffer[]
}
</script>
