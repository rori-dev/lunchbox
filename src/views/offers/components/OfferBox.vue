<template>
  <b-list-group>
    <b-list-group-item variant='primary'>
      <b>{{ provider.name }}</b>
    </b-list-group-item>
    <b-list-group-item v-for='offer in offers' :key='offer.id'>
      <b-row>
        <b-col cols='9' class='offer-name'>{{ offer.name }}</b-col>
        <b-col cols='3' class='offer-price'>{{ offer.price | formatEuro }}</b-col>
      </b-row>
    </b-list-group-item>
  </b-list-group>
</template>



<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { LunchOffer, LunchProvider } from '@/model'

@Component
export default class OfferBox extends Vue {
  @Prop() private provider!: LunchProvider

  @Prop() private offers!: LunchOffer[]
}
</script>



<style lang="scss">
.list-group-item-primary {
  text-align: center;
}

.offer-name {
  padding-right: 8px;
}

.offer-price {
  text-align: right;
  white-space: nowrap;
  padding-left: 0;
}
</style>
