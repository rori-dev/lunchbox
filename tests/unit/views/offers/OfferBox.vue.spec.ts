import OfferBox from '@/views/offers/OfferBox.vue'
import { mensa, gyros, soljanka } from '../../test-data'
import { createComponent } from '../../test-util'

describe('OfferBox', () => {
  it('renders title & offers as list items', () => {
    const wrapper = createComponent(OfferBox, {
      provider: mensa,
      offers: [gyros, soljanka],
    })

    const items = wrapper.findAll('b-list-group-item-stub')
    expect(items.length).toEqual(3)
  })

  it('renders title as first list item', () => {
    const wrapper = createComponent(OfferBox, {
      provider: mensa,
      offers: [gyros, soljanka],
    })

    const title = wrapper.findAll('b-list-group-item-stub').at(0)
    expect(title.text()).toEqual(mensa.name)
  })

  it('renders offers in order', () => {
    const wrapper = createComponent(OfferBox, {
      provider: mensa,
      offers: [gyros, soljanka],
    })

    const items = wrapper.findAll('b-list-group-item-stub')

    const offerItem1 = items.at(1)
    expect(offerItem1.find('.offer-name').text()).toEqual('Gyros')
    expect(offerItem1.find('.offer-price').text()).toEqual('3,50 €')

    const offerItem2 = items.at(2)
    expect(offerItem2.find('.offer-name').text()).toEqual('Soljanka')
    expect(offerItem2.find('.offer-price').text()).toEqual('2,50 €')
  })

  it('renders just title WHEN offers are empty', () => {
    const wrapper = createComponent(OfferBox, {
      provider: mensa,
      offers: [],
    })

    const items = wrapper.findAll('b-list-group-item-stub')
    expect(items.length).toEqual(1)
  })

})
