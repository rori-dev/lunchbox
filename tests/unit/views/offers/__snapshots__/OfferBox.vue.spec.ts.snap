// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferBox renders snapshot 1`] = `
<article
  class="flex flex-col items-center
         px-4 py-2 sm:pt-4 sm:pb-6
         shadow-xl bg-white rounded-lg
         border-l-8 sm:border-t-8 sm:border-l-0 border-primary-400"
>
  <h2
    class="text-3xl sm:text-4xl font-serif text-neutral-800"
  >
    
    Mensa
  
  </h2>
   
  <offer-stub
    offer="[object Object]"
  />
  <offer-stub
    offer="[object Object]"
  />
</article>
`;
